src/utils/traffic-mesh.js

```diff
@@ -92,6 +97,65 @@ const startForwardProxy = async ({ port, frameworkPort, functionsPort, publishDi
   }
 }
 
+const forwardMessagesToLog = ({ log, subprocess }) => {
+  let currentId = null
+  let spinner = null
+
+  subprocess.stderr
```

Comment:
>Can we use something like: [...] To avoid using https://github.com/dominictarr/event-stream?

I can do that.

> Also, should we be reading stdout?

Currently mesh-forward outputs the status events to stderr, to be resilient to the `log-file` option being present or not. If it's not present mesh-forward will output tracing logs to stdout which might cause conflicts if we log progress messages at the same time.

I am looking into creating a separate pipe just for those messages. The `spawn` function has the ability to open multiple pipes, but I'm not yet sure how to consume them in a cross-platform way from Rust.

> We should also change the way we spawn the process as we're using stdio: 'inherit'

Fair.

Reply:
Could we also just read the stream as is, with `data` events?

Reply:
I don't think we should depend on the data events coming line-by-line as we get them from Rusts `eprintln!(...)`. If the pipe is buffered for some random reason this will break our code.